[Bug]: Console Error Override Bug

### Bug Description

@claude

The console.error override in index.ts:42-47 contains a critical bug where args.join(' ') can throw an error if arguments contain non-stringifiable objects, circular references, or complex data structures. This could cause the entire application to crash when attempting to log errors.

### Steps to Reproduce

Open src/index.ts
Navigate to lines 42-47
Pass a circular reference to console.error:
typescriptconst obj: any = {};
obj.circular = obj;
console.error("Error:", obj); // Will crash

### Expected Behavior

Console.error should safely handle all types of input without crashing:

Circular references should be handled
Non-stringifiable objects should be converted safely
Errors should be properly formatted

### Actual Behavior

Current implementation:
typescriptconsole.error = (...args: any[]): void => {
  if (args.some(arg => arg instanceof Error)) {
    process.stderr.write(`[ERROR] ${args.join(' ')}\n`); // UNSAFE!
  }
};
This will throw:

TypeError: Converting circular structure to JSON
TypeError: Cannot convert object to primitive value

### Environment

Claude Desktop (macOS)

### Node.js Version

_No response_

### Branch Thinking Version

1.7.1

### MCP Configuration

_No response_

### Error Logs

```shell

```

### Additional Context

Impact:

Severity: High - Can crash the application
Scope: Error handling throughout the application
Risk: Masks real errors and prevents debugging
MCP Protocol: Could corrupt MCP communication

Proposed Solution:
typescriptconsole.error = (...args: any[]): void => {
  if (args.some(arg => arg instanceof Error)) {
    const safeStr = args
      .map(a => {
        if (a instanceof Error) {
          return `${a.name}: ${a.message}`;
        }
        if (typeof a === 'string') {
          return a;
        }
        try {
          return JSON.stringify(a);
        } catch {
          return String(a);
        }
      })
      .join(' ');
    process.stderr.write(`[ERROR] ${safeStr}\n`);
  }
};
Test Cases:
typescript// Should handle without crashing:
console.error("Simple string");
console.error(new Error("Test error"));
console.error("Multiple", "arguments", 123);
console.error({ circular: /* self-reference */ });
console.error(undefined, null, NaN);
console.error(Symbol('test'));
console.error(() => {});
Acceptance Criteria:

 Console.error handles all JavaScript types without crashing
 Circular references are detected and handled
 Error objects are properly formatted
 Non-stringifiable objects are converted safely
 MCP protocol remains uncorrupted
 All existing error logging continues to work

--- a/src/index.ts
+++ b/src/index.ts
@@ -42,7 +42,22 @@ console.debug = (): void => {};
 console.error = (...args: any[]): void => {
   // Only log actual errors to stderr for debugging
   if (args.some(arg => arg instanceof Error)) {
-    process.stderr.write(`[ERROR] ${args.join(' ')}\n`);
+    const safeStr = args
+      .map(a => {
+        if (a instanceof Error) {
+          return `${a.name}: ${a.message}`;
+        }
+        if (typeof a === 'string') {
+          return a;
+        }
+        try {
+          return JSON.stringify(a);
+        } catch {
+          return String(a);
+        }
+      })
+      .join(' ');
+    process.stderr.write(`[ERROR] ${safeStr}\n`);
   }
 };
 

